scripts/who_is_better.coffee

```diff
@@ -0,0 +1,105 @@
+Promise = require('bluebird')
+
+module.exports = (robot) ->
+  durationService = new Duration()
+
+
+  # Duration This Month
+  #
+  robot.respond /(кто молодец в этом месяце)/i, (res) ->
+    durationService.getCount('duration', 'current')
+    .then((result) -> res.send(result))
+    .catch((err) ->
+      robot.logger.error err
+      res.reply 'Я занят, попросите позже'
+    )
+
+  robot.router.post '/hubot/duration_this_month', (req, res) ->
+    durationService.getCount('duration', 'current')
+    .then((result) -> res.json({text: result}))
+    .catch((err) ->
+      robot.logger.error err
+      res.status(500).json({text: 'Я занят, попросите позже'})
+    )
+  # Approved This Month
+  #
+  # robot.respond /(кто самый оплачиваемый)/i, (res) ->
+  #   durationService.getCount('approved_duration', 'current')
+  #   .then((result) -> res.send(result))
+  #   .catch((err) ->
+  #     robot.logger.error err
+  #     res.reply 'Я занят, попросите позже'
+  #   )
+  #
+  # robot.router.post '/hubot/approved_duration_this_month', (req, res) ->
+  #   durationService.getCount('approved_duration', 'current')
+  #   .then((result) -> res.json({text: result}))
+  #   .catch((err) ->
+  #     robot.logger.error err
+  #     res.status(500).json({text: 'Я занят, попросите позже'})
+  #   )
+
+  # Duration Previous Month
+  #
+  robot.respond /(кто молодец в прошлом месяце)/i, (res) ->
+    durationService.getCount('duration', 'previos')
+    .then((result) -> res.send(result))
+    .catch((err) ->
+      robot.logger.error err
+      res.reply 'Я занят, попросите позже'
+    )
+
+  robot.router.post '/hubot/duration_previous_month', (req, res) ->
+    durationService.getCount('duration', 'previos')
+    .then((result) -> res.json({text: result}))
+    .catch((err) ->
+      robot.logger.error err
+      res.status(500).json({text: 'Я занят, попросите позже'})
+    )
+
+  # Approved Previous Month
+  #
+  # robot.respond /(кто был самым оплачиваемый)|(кому мы скажем спасибо за прошлый месяц)|(кто в прошлом месяце принес денег)/i, (res) ->
+  #   durationService.getCount('approved_duration', 'previos')
+  #   .then((result) -> res.send(result))
+  #   .catch((err) ->
+  #     robot.logger.error err
+  #     res.reply 'Я занят, попросите позже'
+  #   )
+  #
+  # robot.router.post '/hubot/approved_duration_previous_month', (req, res) ->
+  #   durationService.getCount('approved_duration', 'previos')
+  #   .then((result) -> res.json({text: result}))
+  #   .catch((err) ->
+  #     robot.logger.error err
+  #     res.status(500).json({text: 'Я занят, попросите позже'})
+  #   )
+
+
+
+class Pretty
+  TTL: 5
+
+  getCount: (field, month) ->
+    time = new Date().valueOf()
+    new Promise((resolve, reject) =>
+      return resolve(@_best_name) if @_lastFetched? && (@_lastFetched + @TTL * 1000) - time > 0
+      request.get("#{@API_URL}?field=#{field}&month=#{month}")
+      .then((response) =>
+        best_name = response.data
```

Comment:
And please avoid variables mutation. Especially in same way, when you're changing variable type, when left side and right side aren't consistent in names.